.github/workflows/ci.yaml

```diff
@@ -114,6 +114,57 @@ jobs:
         run: sudo make build
       - name: test-e2e
         run: sudo make test-e2e-for-fluentd # containerd interaction requires sudo
+  e2e-tests-for-splunk:
+    strategy:
+      fail-fast: false
+      matrix:
+        go: [ '1.20', '1.21' ]
+        os: [ ubuntu-latest ] # TODO: Add Windows e2e tests: https://github.com/aws/shim-loggers-for-containerd/issues/68
+    name: E2E tests / splunk / ${{ matrix.os }} / Go ${{ matrix.go }}
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go }}
+          cache: false
+      - name: install and start containerd
+        shell: bash
+        run: sudo scripts/install-containerd
+      - name: start splunk local endpoint
+        id: splunk-local-endpoint
+        shell: bash
+        run: |
+          SPLUNK_PASSWORD=1234567Aa!
+          SPLUNK_PORT=8089
+          SPLUNK_IMAGE=splunk/splunk
+          CONTAINER_ID=$(docker run -d -p $SPLUNK_PORT:8089 -p 8088:8088 \
+          -e "SPLUNK_START_ARGS=--accept-license" \
+          -e "SPLUNK_PASSWORD=$SPLUNK_PASSWORD" \
+          -e "SPLUNK_HEC_TOKEN=abcd1234" \
+          $SPLUNK_IMAGE)
+          
+          : # Splunk can only receives requests after becoming healthy
+          until [ $(docker inspect $CONTAINER_ID --format ‘{{.State.Health.Status}}’) == ‘healthy’ ]; do sleep 10s; done;
+          
+          curl -L -k -u admin:$SPLUNK_PASSWORD \
+          -X POST https://localhost:$SPLUNK_PORT/services/admin/token-auth/tokens_auth \
+          -d disabled=false
+          : # Wait for token to be enabled
+          sleep 5
```

Comment:
Could this introduce test variability/flakiness?

Reply:
Maybe but 5s is a super large duration from my test.

Reply:
Is there an endpoint you can check to see if the token is available yet or not?

Reply:
No AFAIK. If we find it is flaky later, we can add retry for creating token.